listenbrainz/webserver/static/js/src/user/Listens.tsx

```diff
@@ -774,111 +763,109 @@ export default class RecentListens extends React.Component<
+                      role="button"
+                      onClick={this.handleClickOlder}
+                      onKeyDown={(e) => {
+                        if (e.key === "Enter") this.handleClickOlder();
+                      }}
+                      tabIndex={0}
+                      href={
+                        isOlderButtonDisabled
+                          ? undefined
+                          : `?max_ts=${nextListenTs}`
+                      }
                     >
-                      <a
-                        role="button"
-                        onClick={this.handleClickOlder}
-                        onKeyDown={(e) => {
-                          if (e.key === "Enter") this.handleClickOlder();
-                        }}
-                        tabIndex={0}
-                        href={
-                          isOlderButtonDisabled
-                            ? undefined
-                            : `?max_ts=${nextListenTs}`
-                        }
-                      >
-                        Older &rarr;
-                      </a>
-                    </li>
-                    <li
-                      className={`next ${
-                        isOldestButtonDisabled ? "disabled" : ""
-                      }`}
+                      Older &rarr;
+                    </a>
+                  </li>
+                  <li
+                    className={`next ${
+                      isOldestButtonDisabled ? "disabled" : ""
+                    }`}
+                  >
+                    <a
+                      role="button"
+                      onClick={this.handleClickOldest}
+                      onKeyDown={(e) => {
+                        if (e.key === "Enter") this.handleClickOldest();
+                      }}
+                      tabIndex={0}
+                      href={
+                        isOldestButtonDisabled
+                          ? undefined
+                          : `?min_ts=${oldestListenTs - 1}`
+                      }
                     >
-                      <a
-                        role="button"
-                        onClick={this.handleClickOldest}
-                        onKeyDown={(e) => {
-                          if (e.key === "Enter") this.handleClickOldest();
-                        }}
-                        tabIndex={0}
-                        href={
-                          isOldestButtonDisabled
-                            ? undefined
-                            : `?min_ts=${oldestListenTs - 1}`
-                        }
-                      >
-                        &#x21E5;
-                      </a>
-                    </li>
-                  </ul>
-                )}
+                      &#x21E5;
+                    </a>
+                  </li>
+                </ul>
                 {currentUser && (
                   <PinRecordingModal
                     recordingToPin={recordingToPin || listens[0]}
```

Comment:
I can't comment on the exact line due to Github limitations so writing here. The Review Modal just a  few lines below should be inside the currentUser check because not logged in users cannot write a review. Its that way in the new RecentListens.tsx already.